### PR TITLE
sensor: lis2ds12: Fixup SPI CS handling

### PR DESCRIPTION
**Pull request by galak**
Replace undefined Kconfig SPI CS defines with DTS based defines.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by avisconti — APPROVED** [review]
Yes, correct.
Actually I put a note on my backlog since I wanted to do it later on. I didn't test SPI extensively
and I wanted to prepare a proper h/w setup for it. Thanks for the change!

File: drivers/sensor/lis2ds12/lis2ds12_spi.c [hunk 1]
```diff
@@ -156,25 +156,23 @@ int lis2ds12_spi_init(struct device *dev)
 
 	data->hw_tf = &lis2ds12_spi_transfer_fn;
 
-#if defined(CONFIG_LIS2DS12_SPI_GPIO_CS)
+#if defined(DT_ST_LIS2DS12_0_CS_GPIO_CONTROLLER)
```
[the remark this answers is not in the document]
Comment (galak) [reply drivers/sensor/lis2ds12/lis2ds12_spi.c]: > So, what are really the DT_ defines generated by scripts? Do they have LIS2DS12 internally?

not sure I follow the question.

[the remark this answers is not in the document]
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_spi.c]: @galak 
Don't worry, it was just brainstorming. Now I see that scripts generate correct DT_ names.
Pls consider my comment above just as random noise, sorry.

File: drivers/sensor/lis2ds12/lis2ds12_spi.c [hunk 2]
```diff
@@ -156,25 +156,23 @@ int lis2ds12_spi_init(struct device *dev)
...
-			CONFIG_LIS2DS12_SPI_GPIO_CS_DRV_NAME);
-		if (!lis2ds12_cs_ctrl.gpio_dev) {
-			LOG_ERR("Unable to get GPIO SPI CS device");
-			return -ENODEV;
-		}
+	lis2ds12_cs_ctrl.gpio_dev = device_get_binding(
+		DT_ST_LIS2DS12_0_CS_GPIO_CONTROLLER);
+	if (!lis2ds12_cs_ctrl.gpio_dev) {
+		LOG_ERR("Unable to get GPIO SPI CS device");
+		return -ENODEV;
```
Comment (MaureenHelm) [inline drivers/sensor/lis2ds12/lis2ds12_spi.c]: Doesn't this force using GPIO for CS now, whereas before it was an option?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**